### PR TITLE
Fix incorrect comments in unjoin command

### LINKED ISSUE #4122: Fix incorrect comments in unjoin command
**Task description**:
Fix some incorrect comments found when going through the code for #4120.

**Solution**:
```diff
diff --git a/pkg/karmadactl/unjoin/unjoin.go b/pkg/karmadactl/unjoin/unjoin.go
index feb7ee907..27652fd9d 100644
--- a/pkg/karmadactl/unjoin/unjoin.go
+++ b/pkg/karmadactl/unjoin/unjoin.go
@@ -78,10 +78,10 @@ type CommandUnjoinOption struct {
        // ClusterNamespace holds namespace where the member cluster secrets are stored
        ClusterNamespace string

-       // ClusterName is the cluster's name that we are going to join with.
+       // ClusterName is the cluster's name that we are going to un-join with.
        ClusterName string

-       // ClusterContext is the cluster's context that we are going to join with.
+       // ClusterContext is the cluster's context that we are going to un-join with.
        ClusterContext string

```

**Who can join or take the task**:

The good first issue is intended for `first-time contributors` to get started on his/her contributor journey.

After a contributor has successfully completed 1-2 good first issue's,
they should be ready to move on to `help wanted` items, saving the remaining `good first issue` for other new contributors.

**How to join or take the task**:

Just reply on the issue with the message `/assign` in a separate line.

Then, the issue will be assigned to you.

**How to ask for help**:

If you need help or have questions, please feel free to ask on this issue.
The issue author or other members of the community will guide you through the contribution process.

### PR DESCRIPTION
**Pull request by Yan-Daojiang**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
/kind deprecation
**Which issue(s) this PR fixes**:
Fixes #4122

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/karmadactl/unjoin/unjoin.go [hunk 1]
```diff
@@ -78,10 +78,10 @@ type CommandUnjoinOption struct {
 	// ClusterNamespace holds namespace where the member cluster secrets are stored
 	ClusterNamespace string
 
-	// ClusterName is the cluster's name that we are going to join with.
+	// ClusterName is the cluster's name that we are going to un-join with.
```
Comment (carlory) [inline pkg/karmadactl/unjoin/unjoin.go:81]:
```suggestion
	// ClusterName is the cluster's name that we are going to unjoin with.
```

File: pkg/karmadactl/unjoin/unjoin.go [hunk 2]
```diff
@@ -78,10 +78,10 @@ type CommandUnjoinOption struct {
...
 	ClusterName string
 
-	// ClusterContext is the cluster's context that we are going to join with.
+	// ClusterContext is the cluster's context that we are going to un-join with.
```
Comment (carlory) [inline pkg/karmadactl/unjoin/unjoin.go:84]:
```suggestion
	// ClusterContext is the cluster's context that we are going to unjoin with.
```

**Comment by Yan-Daojiang** [comment]
Hi @carlory ，Done！

**Comment by RainbowMango** [comment]
Hi @Yan-Daojiang The CI (DCO) is failing, you might need to [squash your commits](https://karmada.io/docs/contributor/github-workflow#squash-commits) and push again.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve
Thanks @Yan-Daojiang for your quick response.

**Pull request merged**